Core/src/IfcGeometryConverter/ProfileConverter.h

```diff
@@ -1485,33 +1399,17 @@ namespace OpenInfraPlatform {
 				coords.push_back(carve::geom::VECTOR(-xBottom * 0.5 + xOffset + xTop, y * 0.5));
 				coords.push_back(carve::geom::VECTOR(-xBottom * 0.5 + xOffset, y * 0.5));
 			}
-
-			//Function 12: GetCircleRadius
-			void AddCircleCoordinates(std::vector<carve::geom::vector<2>>& coords, const double radius, const double d_angle, const int num_segments ) const
-			{
-				
-				double angle = 0;
-				for (int i = 0; i < num_segments; ++i) {
-					coords.push_back(carve::geom::VECTOR((radius * cos(angle)), (radius * sin(angle))));
-					angle += d_angle;
-				}
-				
-			}
-			
-			//Function 13: GetAsymmetricIShapeCoordinates
-			void AddAsymmetricIShapeCoordinates(std::vector<carve::geom::vector<2>>& coords, const double radius, const double Webthickness, const double FlangeThickness, const double h) const
+			// Function 12: Select addArc or push.back
+			void addArc_or_push_back(std::vector<carve::geom::vector<2>>& coords, double  radius, double startAngle, double openingAngle, double xM, double yM, double push_back_x, double push_back_y, int numSegments = -1) const
 			{
-				if (radius != 0) {
-					addArc(coords, radius, M_PI, -M_PI_2, Webthickness * 0.5 + radius, h * 0.5 - FlangeThickness - radius);
+				if (radius != 0.0) {
+					addArc(coords, radius, startAngle, openingAngle, xM, yM);
 				}
 				else {
-					coords.push_back(carve::geom::VECTOR(Webthickness * 0.5, (h * 0.5 - FlangeThickness)));
+					coords.push_back(carve::geom::VECTOR(push_back_x, push_back_y));
 				}
 			}

```

Comment:
This function seems finished. Doxy-comment, please.

Reply:
Done